tests/tox_env/test_api.py

```diff
@@ -0,0 +1,21 @@
+import os
+from pathlib import Path
+
+from tox.pytest import ToxProjectCreator
+
+
+def test_ensure_temp_dir_exists(tox_project: ToxProjectCreator) -> None:
+    ini = "[testenv]\ncommands=python -c 'print(1)'"
+    project = tox_project({"tox.ini": ini})
+    result = project.run()
+    result.assert_success()
+    assert result.state.conf.core["temp_dir"].exists()
+
+
+def test_dont_cleanup_temp_dir(tox_project: ToxProjectCreator, tmp_path: Path) -> None:
+    ini = "[tox]\ntemp_dir={work_dir}/foo\n[testenv]\ncommands=python -c 'print(1)'"
+    (tmp_path / "foo" / "bar").mkdir(parents=True)
+    project = tox_project({"tox.ini": ini})
+    result = project.run("--workdir", os.fspath(tmp_path))
```

Comment:
why do we need workdir here? shouldn't just be the test above but ensure a file in the temp_dir exists before and after the call?

Reply:
I tried that, but on windows conf[temp_dir] was a dot path and always failed

Reply:
That doesn't seem right, I will need to look into it.